client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -115,7 +119,7 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 			recordTracksEvent( 'calypso_signup_unified_design_picker_view', {
 				vertical_id: siteVerticalId,
 				generated_designs: generatedDesigns.map( ( design ) => design.slug ).join( ',' ),
-				static_designs: staticDesigns.map( ( design ) => design.slug ).join( ',' ),
+				has_vertical_images: siteSettings.site_has_vertical_images,
```

Comment:
Since the API returns generated designs only when:
1. The site vertical is whitelisted.
2. The site vertical has enough images.

We can just set `has_vertical_images: generatedDesigns.length > 0` and avoid saving `has_vertical_images` as an site option. What do you think? cc: @autumnfjeld

Reply:
It will simplify the logic, thanks for that!